code/game/objects/items/devices/mat_synth.dm

```diff
@@ -106,97 +103,81 @@ var/static/list/mat2type = list(
+		else
+			to_chat(user, "<span class='warning'>\The [src] matter is not enough to create the selected material!</span>")
+	else if (matter >= 1)
+		to_chat(user, "<span class='warning'>You must select a sheet type first!</span>")
+	else
+		to_chat(user, "<span class='warning'>\The [src] is empty!</span>")
 
 
+/obj/item/device/material_synth/robot/create_material(mob/user, var/material)
+	if(isrobot(user))
+		var/mob/living/silicon/robot/R = user
+		if(R && R.cell)
+			if(R.cell.charge)
+				var/obj/item/stack/sheet/material_type = material
+				if(material_type)
+					var/modifier = get_mat_cost(initial(active_material.perunit))
+					var/amount = input(user, "How many sheets of [initial(material_type.name)] do you want to synthesize? (0 - 50)", "Material Synthesizer") as num
+					amount = clamp(round(amount, 1), 0, 50)
+					if(amount)
+						if(TakeCost(amount, modifier, R))
+							var/obj/item/stack/sheet/inside_sheet = (locate(material_type) in R.module.modules)
+							var/newsheet = FALSE
+							if(!inside_sheet)
+								inside_sheet = new material_type(R.module)
+								R.module.modules += inside_sheet
+								newsheet = TRUE
+							if((inside_sheet.amount + (amount*(!newsheet))) <= inside_sheet.max_amount) // only add amount if no new sheet
+								inside_sheet.amount += amount-(inside_sheet.amount*newsheet) // make equal to amount if no new sheet
+								to_chat(R, "<span class='notice'>Added [amount] of [initial(material_type.name)] to the stack.</span>")
+							else
+								if(inside_sheet.amount <= inside_sheet.max_amount)
+									var/transfer_amount = min(inside_sheet.max_amount - inside_sheet.amount, amount)
+									inside_sheet.amount += (transfer_amount-newsheet) // minus one if new sheet
+									amount -= transfer_amount
+								if(amount >= 1 && (inside_sheet.amount >= inside_sheet.max_amount))
+									to_chat(R, "<span class='warning'>Dropping [amount] sheets, you cannot hold anymore [initial(material_type.name)].</span>")
+									var/obj/item/stack/sheet/dropped_sheet = new material_type(get_turf(src))
+									dropped_sheet.amount = amount-newsheet // minus one if new sheet
+							R.module.rebuild()
+							R.hud_used.update_robot_modules_display()
+						else
+							to_chat(R, "<span class='warning'>You can't make that much [initial(material_type.name)] without shutting down!</span>")
+				else
+					to_chat(R, "<span class='warning'>You need to select a sheet type first!</span>")
 			else
-				to_chat(user, "<span class='warning'>\The [src] matter is not enough to create the selected material!</span>")
-				return
-		else if (matter >= 1)
-			to_chat(user, "<span class='warning'>You must select a sheet type first!</span>")
-			return
+				to_chat(R, "<span class='warning'>Your cell has no charge to use this with!</span>")
 		else
-			to_chat(user, "<span class='warning'>\The [src] is empty!</span>")
-
-	return 1
+			to_chat(R, "<span class='warning'>Your need a cell to use this!</span>")
```

Comment:
Your need a cell to use this! :eyes:

Reply:
it's a borg one and the if above the else checks for a cell

Reply:
"Your" need a cell to use this, it's a grammar issue

Reply:
ah, fixed